src/libraries/System.Console/tests/ReadAndWrite.cs

```diff
@@ -261,22 +262,30 @@ private static unsafe void ValidateConsoleEncoding(Encoding encoding)
     [Fact]
     public static unsafe void OutputEncodingPreamble()
     {
-        Encoding curEncoding = Console.OutputEncoding;
-
-        try
+        try 
         {
-            Encoding encoding = Console.Out.Encoding;
-            // The primary purpose of ConsoleEncoding is to return an empty preamble.
-            Assert.Equal(Array.Empty<byte>(), encoding.GetPreamble());
+            Encoding curEncoding = Console.OutputEncoding;
 
-            // Try setting the ConsoleEncoding to something else and see if it works.
-            Console.OutputEncoding = Encoding.Unicode;
-            // The primary purpose of ConsoleEncoding is to return an empty preamble.
-            Assert.Equal(Array.Empty<byte>(), Console.Out.Encoding.GetPreamble());
+            try
+            {
+                Encoding encoding = Console.Out.Encoding;
+                // The primary purpose of ConsoleEncoding is to return an empty preamble.
+                Assert.Equal(Array.Empty<byte>(), encoding.GetPreamble());
+
+                // Try setting the ConsoleEncoding to something else and see if it works.
+                Console.OutputEncoding = Encoding.Unicode;
+                // The primary purpose of ConsoleEncoding is to return an empty preamble.
+                Assert.Equal(Array.Empty<byte>(), Console.Out.Encoding.GetPreamble());
+            }
+            finally
+            {
+                Console.OutputEncoding = curEncoding;
+            }
         }
-        finally
+        catch(PlatformNotSupportedException)
         {
-            Console.OutputEncoding = curEncoding;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("BROWSER")))
+                Assert.True(true);
         }
     }
```

Comment:
If we're expecting this to fail with PNSE because the functionality isn't supported, we should just disable the test on the relevant platform via attributes.

Reply:
Is it also true for the case when some functionality will be implemented later?

Reply:
Yes. If it will be implemented later, then there should be an issue tracking implementing it, and the test should be disabled with [ActiveIssue(http://.../issuenumber)].

Reply:
Thanks!

Reply:
> If we're expecting this to fail with PNSE because the functionality isn't supported, we should just disable the test on the relevant platform via attributes

@stephentoub Are you sure we want to do that? How do you check today that PNSE is actually thrown and it does not throw NRE or some other exception instead?

Reply:
We can have one or more dedicated tests that validate the PNSE behavior, e.g.
```C#
[PlatformSpecific(...)]
public void ValidateAllMethodsThrowPNSEOnBrowser()
{
    Assert.Throws<PlatformNotSupportedException>(() => Console.Foo);
    Assert.Throws<PlatformNotSupportedException>(() => Console.Bar);
    ...
}
```
We should not add try/catch logic to every single test just to handle a platform not supporting it.

Reply:
I've added a separate wasm-specific test to handle PNSE. Is it correct now?

Reply:
One comment on what it's testing, but the general approach looks fine.